src/shogun/distance/MahalanobisDistance.cpp

```diff
@@ -38,21 +36,25 @@ CMahalanobisDistance::~CMahalanobisDistance()
 bool CMahalanobisDistance::init(CFeatures* l, CFeatures* r)
 {
 	CRealDistance::init(l, r);
-
+	SGMatrix<float64_t> cov;
 
 	if ( l == r)
 	{
 		mean = ((CDenseFeatures<float64_t>*) l)->get_mean();
-		icov  = ((CDenseFeatures<float64_t>*) l)->get_cov();
+		cov = ((CDenseFeatures<float64_t>*)l)->get_cov();
```

Comment:
while you are changing these lines, could you please replace these explicit casts to `dynamic_cast` at least plz? :)
or even better do a `REQUIRED` assertation for the type